# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/libv4l/hotfix-gcc14.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- v4l-utils-1.28.0/lib/libv4lconvert/jl2005bcd.c.vanilla	2024-07-23 12:14:22.830730996 +0200
+++ v4l-utils-1.28.0/lib/libv4lconvert/jl2005bcd.c	2024-07-23 12:14:41.432729204 +0200
@@ -63,7 +63,7 @@
 	struct jpeg_decompress_struct dinfo;
 	struct jpeg_error_mgr jcerr, jderr;
 	JOCTET *jpeg_header = NULL;
-	unsigned long jpeg_header_size = 0;
+	size_t jpeg_header_size = 0;
 	int i, x, y, x1, y1, jpeg_data_size, jpeg_data_idx, eoi, size;
 
 	/* src_size had better be bigger than 16 */
--- v4l-utils-1.28.0/lib/libv4lconvert/jpeg.c.vanilla	2024-07-23 12:15:09.409726508 +0200
+++ v4l-utils-1.28.0/lib/libv4lconvert/jpeg.c	2024-07-23 12:15:22.447725252 +0200
@@ -136,7 +136,7 @@
 {
 	struct jpeg_compress_struct cinfo;
 	unsigned char *jpeg_header = NULL;
-	unsigned long jpeg_header_size = 0;
+	size_t jpeg_header_size = 0;
 
 	if (data->cinfo_initialized)
 		return;
